File: AdvantageFramework.Core/Database/Entities/ClientPortalAlertRecipient.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdvantageFramework.Core.Database.Entities
{
    public partial class ClientPortalAlertRecipient
    {
        [Key]
        [Column("ALERT_ID")]
        public int AlertId { get; set; }
        [Key]
        [Column("ALERT_RCPT_ID")]
        public int AlertRcptId { get; set; }
        [Column("CDP_CONTACT_ID")]
        public int CdpContactId { get; set; }
        [Column("EMAIL_ADDRESS")]
        [StringLength(50)]
        public string EmailAddress { get; set; }
        [Column("PROCESSED", TypeName = "smalldatetime")]
        public DateTime? Processed { get; set; }
        [Column("NEW_ALERT")]
        public short? NewAlert { get; set; }
        [Column("READ_ALERT")]
        public short? ReadAlert { get; set; }
        [Column("CURRENT_RCPT")]
        public short? CurrentRcpt { get; set; }
        [Column("CURRENT_NOTIFY")]
        public int? CurrentNotify { get; set; }
        [Column("CS_IS_REVIEWER")]
        public bool? CsIsReviewer { get; set; }
        [Column("IS_DELETED")]
        public bool? IsDeleted { get; set; }
        [Column("PROOFING_STATUS_ID")]
        public int? ProofingStatusId { get; set; }

        [ForeignKey(nameof(AlertId))]
        [InverseProperty("ClientPortalAlertRecipients")]
        public virtual Alert Alert { get; set; }
    }
}

File: AdvantageFramework/Database/Entities/ClientPortalAlertRecipient.vb
Namespace Database.Entities

	<Table("CP_ALERT_RCPT")>
	Public Class ClientPortalAlertRecipient
		Inherits BaseClasses.Entity

#Region " Constants "



#End Region

#Region " Enum "

        Public Enum Properties

            AlertID
            AlertRecipientID
            ClientContactID
            EmailAddress
            ProcessedDate
            IsNewAlert
            HasBeenRead
            IsConceptShareReviewer
            IsDeleted
            IsCurrentRecipient
            Alert

        End Enum

#End Region

#Region " Variables "



#End Region

#Region " Properties "

        <Key>
		<Required>
        <Column("ALERT_ID", Order:=0)>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property AlertID() As Integer
		<Key>
		<Required>
        <Column("ALERT_RCPT_ID", Order:=1)>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property AlertRecipientID() As Integer
		<Required>
		<Column("CDP_CONTACT_ID")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property ClientContactID() As Integer
		<MaxLength(50)>
		<Column("EMAIL_ADDRESS", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property EmailAddress() As String
		<Column("PROCESSED")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property ProcessedDate() As Nullable(Of Date)
		<Column("NEW_ALERT")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property IsNewAlert() As Nullable(Of Short)
		<Column("READ_ALERT")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property HasBeenRead() As Nullable(Of Short)
        <Column("CS_IS_REVIEWER")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property IsConceptShareReviewer() As Nullable(Of Boolean)
        <Column("IS_DELETED")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property IsDeleted() As Nullable(Of Boolean)
        <Column("CURRENT_RCPT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property IsCurrentRecipient() As Nullable(Of Short)


        Public Overridable Property Alert As Database.Entities.Alert

#End Region

#Region " Methods "

        Public Overrides Function ToString() As String

            ToString = Me.AlertID

        End Function

#End Region

	End Class

End Namespace
